src/services/inlineHints.ts

```diff
@@ -0,0 +1,300 @@
+        }
+
+        function shouldCallExpressionHint(call: CallExpression) {
+            if (!callExpressionHintableCache.has(call)) {
+                let current = call;
+                while (true) {
+                    if (!isAccessExpression(current.parent) || !isCallExpression(current.parent.parent)) {
+                        callExpressionHintableCache.set(current, false);
+                        break;
+                    }
+
+                    callExpressionHintableCache.set(current, true);
+                    current = current.parent.parent;
+                }
+            }
+            return callExpressionHintableCache.get(call);
+        }
+
+        function makeEmptyRange(position: number): TextSpan {
+            return { start: position, length: 0 };
+        }
+
+        function visitEnumMember(member: EnumMember) {
+            if (member.initializer) {
+                return;
+            }
+
+            const enumValue = checker.getConstantValue(member);
+            if (enumValue !== undefined) {
+                addEnumMemberValueHints(enumValue.toString(), makeEmptyRange(member.end));
+            }
+        }
+
+        function visitVariableLikeDeclaration(decl: VariableDeclaration | PropertyDeclaration) {
+            if (decl.type || !decl.initializer) {
+                return;
+            }
+
+            const declarationType = checker.getTypeAtLocation(decl);
+            if (!preferences.includeInlineRequireAssignedVariableTypeHints && declarationType.symbol && (declarationType.symbol.flags & SymbolFlags.Module)) {
+                return;
+            }
+
+            const typeDisplayString = printTypeInSingleLine(declarationType);
+            if (typeDisplayString) {
+                addTypeHints(typeDisplayString, makeEmptyRange(decl.name.end));
+            }
+        }
+
+        function visitCallOrNewExpression(expr: CallExpression | NewExpression) {
+            if (!expr.arguments || !expr.arguments.length) {
+                return;
+            }
+
+            const candidates: Signature[] = [];
+            const signature = checker.getResolvedSignatureForSignatureHelp(expr, candidates);
+            if (!signature || !candidates.length) {
+                return;
+            }
+
+            for (let i = 0; i < expr.arguments.length; ++i) {
+                const arg = expr.arguments[i];
+                if (!preferences.includeInlineNonLiteralParameterNameHints && !isHintableExpression(arg)) {
+                    continue;
+                }
+
+                const parameterName = checker.getParameterIdentifierNameAtPosition(signature, i);
+                if (parameterName) {
+                    if (preferences.includeInlineDuplicatedParameterNameHints || !isIdentifier(arg) || arg.text !== parameterName) {
+                        addNameHints(unescapeLeadingUnderscores(parameterName), makeEmptyRange(expr.arguments[i].getStart()));

```

Comment:
I'd just capture `expr.arguments`.

```suggestion
            const args = expr.arguments;
            if (!args || !args.length) {
                return;
            }

            const candidates: Signature[] = [];
            const signature = checker.getResolvedSignatureForSignatureHelp(expr, candidates);
            if (!signature || !candidates.length) {
                return;
            }

            for (let i = 0; i < args.length; ++i) {
                const arg = args[i];
                if (!preferences.includeInlineNonLiteralParameterNameHints && !isHintableExpression(arg)) {
                    continue;
                }

                const parameterName = checker.getParameterIdentifierNameAtPosition(signature, i);
                if (parameterName) {
                    if (preferences.includeInlineDuplicatedParameterNameHints || !isIdentifier(arg) || arg.text !== parameterName) {
                        addNameHints(unescapeLeadingUnderscores(parameterName), makeEmptyRange(args[i].getStart()));
```